.github/workflows/ci-nix.yml

```diff
@@ -126,6 +126,17 @@ jobs:
         if: (github.event_name != 'pull_request' || matrix.build-in-pr) && matrix.run-tests && (matrix.host != 'macos')
         run: nix build -L .#wasm32-unknown.ci.wasmTest --keep-failed
 
+      - name: Cleanup disk space
+        if: (github.event_name != 'pull_request' || matrix.build-in-pr) && matrix.run-tests
+        run: sudo rm -rf /usr/local/lib/android
```

Comment:
The buildjet runner started running out of space during the backwards compatibility tests. It's not obvious to me why this just started happening after several successful CI runs previously without a rebase. In a separate branch I identified some low hanging fruit to cleanup, along with deleting the `devimint-*` directory, which was adding ~2GB per matrix iteration.

https://github.com/fedimint/fedimint/pull/4050

https://github.com/fedimint/fedimint/actions/runs/7554418498/job/20567202313?pr=4050#step:11:4336

```
Filesystem      Size  Used Avail Use% Mounted on
udev             14G     0   14G   0% /dev
tmpfs           2.8G  992K  2.8G   1% /run
/dev/sda1       114G  112G  1.6G  99% /
tmpfs            14G  4.0K   14G   1% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs            14G     0   14G   0% /sys/fs/cgroup
/dev/sda15      105M  6.1M   99M   6% /boot/efi
/dev/loop0       64M   64M     0 100% /snap/core20/1822
/dev/loop1       50M   50M     0 100% /snap/snapd/17950
/dev/loop2       92M   92M     0 100% /snap/lxd/24061
tmpfs           2.8G     0  2.8G   0% /run/user/0
/dev/loop3       64M   64M     0 100% /snap/core20/2105
Did not have permissions for all directories
6.8G       ┌── CodeQL          │███▒▒░                                    │   6%
 10G     ┌─┴ hostedtoolcache   │█████░                                    │  10%
 13G   ┌─┴ opt                 │██████                                    │  12%
 11G   │               ┌── deps│█████▓▓▓                                  │  10%
 17G   │             ┌─┴ debug │███████▓                                  │  17%
 17G   │           ┌─┴ target  │███████▓                                  │  17%
 17G   │         ┌─┴ fedimint  │███████▓                                  │  17%
 17G   │       ┌─┴ fedimint    │███████▓                                  │  17%
 17G   │     ┌─┴ _work         │███████▓                                  │  17%
 18G   │   ┌─┴ actions-runner  │████████                                  │  17%
 19G   │ ┌─┴ ubuntu            │████████                                  │  19%
 20G   ├─┴ home                │████████                                  │  19%
 31G   │ ┌── store             │█████████████                             │  30%
 31G   ├─┴ nix                 │█████████████                             │  30%
6.2G   │ ┌── lib               │███░░░░░░░░░░░                            │   6%
 12G   │ │     ┌── sdk         │█████▓▒▒▒░░░░░                            │  11%
 12G   │ │   ┌─┴ android       │█████▓▒▒▒░░░░░                            │  11%
 13G   │ │ ┌─┴ lib             │██████▒▒▒░░░░░                            │  13%
 22G   │ ├─┴ local             │█████████░░░░░                            │  21%
 36G   ├─┴ usr                 │██████████████                            │  34%
106G ┌─┴ /                     │█████████████████████████████████████████ │ 100%
```

Reply:
@bradleystachurski The problem is that `cargo build` etc. is very wasteful.

Nix makes it better because `crane` compresses `./target` directory and keeps it deduplicated between runs.

Now this change introduces a whole new build of `./target`, just like a developer would have locally, and that does not decompose with with anything else.

We could avoid it, by making these scripts a little smarter and use binaries built already via Nix, instead of running an extra raw `cargo build`.

This can be made conditional by the `CI` env var , that would skip all `cargo build`s, and for the "testing current version" use `nix build` to get the binaries of the current source code.

Reply:
Eh... but this won't work for running tests ... Oh dear ...

In that case maybe running the whole thing on a separate GHA workflow, that runs *after* the main one passes is the way to go.

Reply:
Fixing this might require twisting the solution inside out, so I'm not sure if there's a point landing it...

We might want to think a bit deeper about the whole thing. :/